crates/jstz_node/src/services/logs/db.rs

```diff
@@ -0,0 +1,83 @@
+use super::Line;
+use actix_web::web::block;
+use anyhow::{anyhow, Result};
+use jstz_proto::{js_logger::LogRecord, request_logger::RequestEvent};
+use load_file::load_str;
+use r2d2::{Pool, PooledConnection};
+use r2d2_sqlite::SqliteConnectionManager;
+use std::path::Path;
+pub type SqliteConnectionPool = r2d2::Pool<SqliteConnectionManager>;
+pub type SqliteConnection = r2d2::PooledConnection<r2d2_sqlite::SqliteConnectionManager>;
+
+pub struct DB {
+    pool: SqliteConnectionPool,
+}
+
+impl DB {
+    // Initialize the sql databse by createing a connection pool.
+    // if the database does not exist, it will be created.
+    pub async fn init<P: AsRef<Path>>(path: P) -> Result<Self> {
+        let manager = SqliteConnectionManager::file(path);
+        let pool = SqliteConnectionPool::new(manager)?;
+
+        Self::create(pool.clone()).await?;
```

Comment:
This is related to [this comment](https://github.com/trilitech/jstz/pull/329#discussion_r1469776754). With a reference, a lifetime issue arises in the async move closure in `connection` function.  
So we could either
1. take a simpler approach where we just use pool.get (whcih is blocking)  and pass the pool reference around without cloning
2. Clone the pool so we could connect to the db in a thread where blocing is acceptible. 

I sticked to the second option because pool.get has a loop that blocks the task and cloning a pool itself shouldn't be that costly (its a wrapper around Arc)

wdyt?